dlrover/python/master/node/dist_job_manager.py

```diff
@@ -790,6 +790,34 @@ def _process_event(self, event: NodeEvent):
             should_relaunch = self._should_relaunch(
                 cur_node, status_change_flow
             )
+
+            # Even _should_relaunch return True by state-machine check,
+            # we need another round of check
+            if should_relaunch:
```

Comment:
Move these codes into the method `_should_relaunch` and add UT.

Reply:
it is a k8s plugin going to open sourced

Reply:
ok